datafusion/core/src/physical_planner.rs

```diff
@@ -1757,6 +1757,12 @@ impl DefaultPhysicalPlanner {
             )));
         }
 
+        if !e.logical_optimization_succeeded {
+            if let Some(plan) = e.stringified_plans.last() {
```

Comment:
I think the original intent was to still show the explain plan even if there was an error 

The idea was to show all the work that had happened so far (e.g. the plans) to help debugging

I agree both explain modes should be consistent, but I suggest that instead of erroring with `EXPLAIN FORMAT INDENT` instead we capture and report the error in `EXPLAIN FORMAT TREE`